pandas/core/indexing.py

```diff
@@ -85,19 +86,14 @@ class IndexingError(Exception):
     pass
 
 
-class _NDFrameIndexer(object):
+class _NDFrameIndexer(_NDFrameIndexerBase):
     _valid_types = None
     _exception = KeyError
     axis = None
 
-    def __init__(self, obj, name):
-        self.obj = obj
-        self.ndim = obj.ndim
-        self.name = name
-
     def __call__(self, axis=None):
         # we need to return a copy of ourselves
```

Comment:
did you reverse these on purpose?

Reply:
Yes, so that we can use `functools.partial(indexer, name)` to shave a few % more. If that's not important, I can revert to the original order.

Reply:
ok this is not a big deal to reverse.